src/Workspaces/Core/Portable/Serialization/SerializerService.cs

```diff
@@ -109,12 +108,6 @@ public void Serialize(object value, ObjectWriter writer, SolutionReplicationCont
             {
                 cancellationToken.ThrowIfCancellationRequested();
 
-                if (value is ChecksumWithChildren checksumWithChildren)
-                {
-                    SerializeChecksumWithChildren(checksumWithChildren, writer, cancellationToken);
-                    return;
-                }
```

Comment:
instead of bucketing all types in a uniform fashoin here, we take care of each type specifically below.